File: src/macos/Sources/AmbientContextCore/Models/MediaSourceKindClassifier.swift
import Foundation

/// SMTC の AppUserModelId / bundle id / 実行ファイル名から、メディアセッションの種別を粗く推定する。
/// 戻り値は `"music"` / `"video"` / `"browser"` / `"unknown"` の 4 種類。
/// `"browser"` は Chrome タブ等で実体が音楽か動画か判定できないケース。
/// 値はヒューリスティック (部分文字列マッチ) であり、誤分類はあり得る前提でクライアント側で扱うこと。
public enum MediaSourceKindClassifier {
    public static func classify(_ sourceApp: String?) -> String {
        guard let sourceApp,
              !sourceApp.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty else {
            return "unknown"
        }

        let lower = sourceApp.lowercased()

        // 動画特化 (Zune は music/video が別 path で存在するので video を先にチェック)
        if lower.contains("zunevideo") ||
            lower.contains("netflix") ||
            lower.contains("primevideo") ||
            lower.contains("amazon.video") ||
            lower.contains("hulu") ||
            lower.contains("disney") ||
            lower.contains("vlc") ||
            lower.contains("mpv") ||
            lower.contains("mpc") {
            return "video"
        }

        // 音楽特化
        if lower.contains("spotify") ||
            lower.contains("itunes") ||
            lower.contains("applemusic") ||
            lower.contains("amazonmusic") ||
            lower.contains("tidal") ||
            lower.contains("zunemusic") ||
            lower.contains("youtubemusic") {
            return "music"
        }

        // ブラウザ (タブの中身が音楽か動画か判定不能)
        if lower.contains("chrome") ||
            lower.contains("msedge") ||
            lower.contains("firefox") ||
            lower.contains("opera") ||
            lower.contains("vivaldi") ||
            lower.contains("brave") {
            return "browser"
        }

        return "unknown"
    }
}

File: src/windows/AmbientContextMcp.Core/Models/MediaSourceKindClassifier.cs
namespace AmbientContextMcp.Core.Models;

/// <summary>
/// SMTC の AppUserModelId / bundle id / 実行ファイル名から、メディアセッションの種別を粗く推定する。
/// 戻り値は <c>"music"</c> / <c>"video"</c> / <c>"browser"</c> / <c>"unknown"</c> の 4 種類。
/// <c>"browser"</c> は Chrome タブ等で実体が音楽か動画か判定できないケース。
/// 値はヒューリスティック (部分文字列マッチ) であり、誤分類はあり得る前提でクライアント側で扱うこと。
/// </summary>
public static class MediaSourceKindClassifier
{
    public static string Classify(string sourceApp)
    {
        if (string.IsNullOrWhiteSpace(sourceApp))
        {
            return "unknown";
        }

        var lower = sourceApp.ToLowerInvariant();

        // 動画特化 (Zune は music/video が別 path で存在するので video を先にチェック)
        if (lower.Contains("zunevideo") ||
            lower.Contains("netflix") ||
            lower.Contains("primevideo") ||
            lower.Contains("amazon.video") ||
            lower.Contains("hulu") ||
            lower.Contains("disney") ||
            lower.Contains("vlc") ||
            lower.Contains("mpv") ||
            lower.Contains("mpc") ||
            lower.Contains("com.apple.tv"))
        {
            return "video";
        }

        // 音楽特化
        if (lower.Contains("spotify") ||
            lower.Contains("itunes") ||
            lower.Contains("applemusic") ||
            lower.Contains("amazonmusic") ||
            lower.Contains("tidal") ||
            lower.Contains("zunemusic") ||
            lower.Contains("youtubemusic") ||
            lower.Contains("com.apple.music") ||
            // Podcasts.app は音声再生専用なので music に寄せる (video/browser のどちらでもない)。
            lower.Contains("com.apple.podcasts"))
        {
            return "music";
        }

        // ブラウザ (タブの中身が音楽か動画か判定不能)
        if (lower.Contains("chrome") ||
            lower.Contains("msedge") ||
            lower.Contains("firefox") ||
            lower.Contains("opera") ||
            lower.Contains("vivaldi") ||
            lower.Contains("brave") ||
            lower.Contains("com.apple.safari") ||
            lower.Contains("com.microsoft.edgemac") ||
            lower.Contains("company.thebrowser.browser"))
        {
            return "browser";
        }

        return "unknown";
    }
}
